### ISSUE TITLE
/1 appended to URL internally, not 301-redirected

### ISSUE THREAD

**Issue by thomaszbz** [dmitryd/typo3-realurl#350]
This issue might rather be a question than a bug report. I just don't know yet.

The issue is that for an URL `https://www.example.com` with TYPO3 CMS 7.6.11 and RealUrl 2.1.5 I get URLs of the form

    https://www.example.com/1
    1

which are then passed via TYPO3's hook in `ContentObjectRenderer` to metaseo and then used as canonical URLs and in sitemaps for search engines.

For details how I tracked this down please have a look at https://github.com/mblaschke/TYPO3-metaseo/issues/224

I am aware that RealUrl silently appends numbers in case of [duplicates](https://github.com/dmitryd/typo3-realurl/wiki/Notes-for-Integrators#my-record-titles-are-not-unique-how-do-i-use-them-in-aliases). I assume that this is different from old issues where URLs had page UIDs appended like `https://www.example.com/44/`. Note that my example hat page `uid=2` and not `1`. So it could be because something duplicates.

However, I could not figure out what was the duplicate in my case. Page title, alias and (after dropping some data) even entries in RealUrl's tables were unique or deleted. The /1 entry got created over and over in RealUrl's tables. Workaround: To stop RealUrl from creating `/1` entries, I had to insert an entry `/`.

Basically, there's two questions:

- When exactly does RealUrl append numbers to URLs? Could you please document how duplicates are determined by RealUrl? Which requirements are needed to ensure uniqueness?
- If such a number is appended internally, should the http-request not be 301-redirected to the internal new URL? That would avoid duplicate content, and be consistent with sitemaps and canonical URLs.

**Comment by dmitryd** [comment]
> I am aware that RealUrl silently appends numbers in case of duplicates

Not, it does not do so for page titles. It does so only for aliases.

> I assume that this is different from old issues where URLs had page UIDs appended like https://www.example.com/44/

This was the case for realurl 1.x and never for 2.x.

> Basically, there's two questions:

This place is for confirmed bugs only. There are no answers to questions here or any kind of support beyond fixing confirmed bugs. If you need other help, see [Getting help](https://github.com/dmitryd/typo3-realurl/wiki/Getting-help).

Before I look into this, you have to make sure this is a realurl bug. So far I doubt so. Therefore I close this issue. If you have more information and can confirm that it is really a realurl bug, please, post more information and I will re-open the report.
